circuits/src/recipient.rs

```diff
@@ -61,6 +59,84 @@ impl Default for RecipientParameters {
     }
 }
 
+impl RecipientParameters {
+    /// Create the recipient parameter
+    pub fn new(
+        rng: &mut (impl RngCore + CryptoRng),
```

Comment:
This is the recommended way of taking an rng by `rand`:
```suggestion
        rng: &mut (impl RngCore + CryptoRng + ?Sized),
```

Reply:
Seems to introduce nasty trait bounds that I don't feel like solving now